manifests/charts/istio-control/istio-discovery/files/injection-template.yaml

```diff
@@ -45,8 +45,8 @@ metadata:
     kubectl.kubernetes.io/default-container: "{{ index $containers 0 }}",
     {{- end }}
     {{- end }}
-{{- if or .Values.pilot.cni.enabled .Values.istio_cni.enabled }}
-    {{- if or (eq .Values.pilot.cni.provider "multus") (eq .Values.istio_cni.provider "multus") (not .Values.istio_cni.chained)}}
+{{- if or .Values.pilot.cniEnabled .Values.istio_cni.enabled }}
```

Comment:
I don't suspect this will be particularly controversial: It should be roughly time to drop `istio_cni` values and associated references?

They were marked as deprecated and slated for removal in https://github.com/istio/istio/pull/49495 which was at least 2 releases ago - I suggest we just ditch them, as I suspect you guys would be the only major impactee either way?

Reply:
Yes, it is only used by us. But I think is better to do this in another PR not to mix changes in the same PR

Reply:
I think we have to remove something in this PR.

I don't want to keep all 3 - current , deprecated, and extra-deprecated - we should never carry more than 1 "deprecated" option.

But if we remove `istio_cni` in this PR, we should be good to simply mark `pilot.cni.xx` as deprecated and move on.

Reply:
This is extremely controversial to me :slightly_smiling_face: . I don't understand why we are doing this at all.

First, we had `.Values.istio_cni.enabled`. Then we decided we didn't like the name, so we made it `pilot.cni.enabled`. Now we are deciding we don't like the new name, so are renaming it **again** to `pilot.cniEnabled`?

The PR says "avoid failures when values are unwrapped", what are the failures? This seems like it should work fine.

Unless there is a real failure unavoidable based on the name, which I suspect is not the case, we should not change this... we cannot rename APIs every few weeks based on our current naming preferences

Reply:
We are doing it for the same reasons we do anything with our charts - the way we did it before caused problems.

> First, we had .Values.istio_cni.enabled. Then we decided we didn't like the name

`istio_cni` values have been explicitly deprecated for at least two releases -> removing them shouldn't be controversial, if the original deprecation marking wasn't.

> so we made it pilot.cni.enabled. Now we are deciding we don't like the new name

I like it just fine, but the problem is, [as you pointed out](https://github.com/istio/istio/pull/52459#discussion_r1700439626), that it has inherent scoping conflicts.

We can live with the scoping conflicts, but actually looking at the values in that scope (there are only two) it's pretty clear one of them (`provider`) is functionally and practically redundant with another value `platform` in every single usage we currently have/test/document.

Which leaves us with just one problematic scoping `cni.enabled` -> as mentioned, if we composed charts we wouldn't need this either, and the entire key that causes scoping issues and renaming controversies would simply vanish as a vestigial thing.

But in the short term our options are

- Keep the problematic scoping, just for this one variable (`pilot.cni.enabled`) (and maybe drop it if we ever compose charts)
- Rename the one variable to a non-problematic scoping `pilot.cniEnabled` (and maybe drop it if we ever compose charts)

I am OK with either of those two, but seeing as the only people that use this are the ones opening the PR, I'm fine fixing the problematic scoping if they are.